x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/failurestore/FailureStoreSecurityRestIT.java

```diff
@@ -1388,12 +1389,8 @@ public void testDlsFls() throws Exception {
             Map.of(dataIndexName, Set.of("@timestamp", "age"))
         );
 
-        // FLS sort of applies to failure store
-        // TODO this will change with FLS handling
-        assertSearchResponseContainsExpectedIndicesAndFields(
-            performRequest(user, new Search("test1::failures").toSearchRequest()),
-            Map.of(failureIndexName, Set.of("@timestamp"))
-        );
+        // FLS does not apply to failure store
+        expectFlsDlsError(() -> performRequest(user, new Search("test1::failures").toSearchRequest()));
```

Comment:
I've hit the ready button too soon. I still want to test cover the case when users have direct access to `.fs-*` indices and accessing without `::failures` selector.